src/writer/serve.py

```
@@ -278,13 +278,13 @@ async def import_zip(file: UploadFile = File(...)):
                     if size > MAX_FILE_SIZE:
                         tmp.close()
                         os.unlink(tmp.name)
-                        raise HTTPException(status_code=413, detail=f"File too large. Max file size: {MAX_FILE_SIZE}")
+                        raise HTTPException(status_code=413, detail={"summary": f"File too large. Max file size: {MAX_FILE_SIZE}"})
                     tmp.write(chunk)
                 tmp_path = tmp.name
             await app_runner.import_zip(tmp_path)
             os.remove(tmp_path)
         except ValueError:
-            raise HTTPException(status_code=400, detail="Invalid upload.")
+            raise HTTPException(status_code=400, detail={"summary": "Invalid archive contents", "details": traceback.format_exc()})
```

Ruff (0.14.14)

[warning] Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)